app/cdap/components/ConnectionList/Components/TabLabelCanSample/index.tsx

```diff
@@ -0,0 +1,124 @@
+import { ConnectionsContext } from 'components/Connections/ConnectionsContext';
+import * as React from 'react';
+import { Redirect } from 'react-router';
+import { getCurrentNamespace } from 'services/NamespaceStore';
+import { useStyles } from './styles';
+
+const TabLabelCanSample = ({
+  label,
+  entity,
+  initialConnectionId,
+  toggleLoader,
+  setIsErrorOnNoWorkSpace,
+}: {
+  label: string;
+  entity: any;
+  initialConnectionId: string;
+  toggleLoader: (value: boolean, isError?: boolean) => void;
+  setIsErrorOnNoWorkSpace: React.Dispatch<React.SetStateAction<boolean>>;
+}) => {
+  const classes = useStyles();
+
+  const myLabelRef: any = React.createRef();
+  const [refValue, setRefValue] = React.useState(false);
+  const [workspaceId, setWorkspaceId] = React.useState(null);
+  const [currentConnection, setCurrentConnection] = React.useState(initialConnectionId);
+
+  const { onWorkspaceCreate } = React.useContext(ConnectionsContext);
+
+  React.useEffect(() => {
+    setRefValue(myLabelRef?.current?.offsetWidth < myLabelRef?.current?.scrollWidth);
+  }, []);
+
+  const onExplore = (entity) => {
+    const { canBrowse, canSample } = entity;
+    if (!canBrowse && canSample) {
+      onCreateWorkspace(entity);
+    } else {
+      setIsErrorOnNoWorkSpace(true);
+    }
+  };
+
+  const onCreateWorkspace = async (entity, parseConfig = {}) => {
+    try {
+      createWorkspaceInternal(entity, parseConfig);
+    } catch (e) {
+      setIsErrorOnNoWorkSpace(true);
+    }
+  };
+
+  const createWorkspaceInternal = async (entity, parseConfig = {}) => {
+    toggleLoader(true);
+    createWorkspace({
+      entity,
+      connection: currentConnection,
+      properties: parseConfig,
+    })
+      .then((res) => {
+        if (onWorkspaceCreate) {
+          return onWorkspaceCreate(res);
+        }
+        if (res) {
+          setWorkspaceId(res);
+          toggleLoader(false);
+        }
+      })
+      .catch((err) => {
+        toggleLoader(false, true);
+        setIsErrorOnNoWorkSpace(true);
+      });
+  };
+
+  return workspaceId ? (
+    <Redirect to={`/ns/${getCurrentNamespace()}/wrangler-grid/${workspaceId}`} />
+  ) : refValue ? (
+    <CustomTooltip title={label} arrow>
+      <Box className={classes.labelsContainerCanSample}>
+        <Typography variant="body1" className={classes.labelStylesCanSample} ref={myLabelRef}>
+          {label}
+        </Typography>
+        <div onClick={() => onExplore(entity)}>
+          <Box className="wranglingHover">
+            <WrangelIcon />
+            <Typography color="primary">Wrangle</Typography>
+          </Box>
+        </div>
+      </Box>
+    </CustomTooltip>
+  ) : (
+    <Box className={classes.labelsContainerCanSample}>
+      <Typography variant="body1" className={classes.labelStylesCanSample} ref={myLabelRef}>
+        {label}
+      </Typography>
+      <div onClick={() => onExplore(entity)}>
```

Comment:
again here don't use a div for a click handler

Reply:
Done